lib/phantomjs/platform.rb

```diff
@@ -102,7 +104,11 @@ def package_url
     class Linux32 < Platform
       class << self
         def useable?
-          host_os.include?('linux') and (architecture.include?('x86_32') or architecture.include?('i686'))
+          host_os.include?('linux') and architecture =~ /i[3-6]86/
+        end
+
+        def warn
+          STDERR.puts "WARNING! Using i686, but Ruby was compiled for #{architecture}. You can check your actual architecture with uname -m." if architecture =~ /i[3-5]86/
```

Comment:
This method overwrites `Kernel.warn`, which ironically enough, is probably what you want instead of writing to stderr: http://ruby-doc.org/core-2.2.3/Kernel.html#method-i-warn